src/main/java/uk/co/ramp/distribution/DistributionSampler.java

```diff
@@ -48,4 +49,9 @@ public int getDistributionValue(Distribution distribution) {
 
     return value;
   }
+
+  public EnumeratedIntegerDistribution resampleDays(int[] outcomes, double[] timeSpread) {
+
+    return new EnumeratedIntegerDistribution(rng.getRandomGenerator(), outcomes, timeSpread);
```

Comment:
Could we return a sampled value here instead of returning the distribution?

Reply:
wouldn't that mean we would be creating a new distribution for each call in the stream? I thought this would be more efficient, but that was more on gut.

Reply:
That’s right, seems like no harm in creating a new distribution object each time though. We would be reusing the same random generator. From an interface point of view it seems to make more sense to return a samples value here to match the return values in all the other methods of this class.